src/cargo/sources/path.rs

```diff
@@ -404,6 +404,32 @@ impl<'gctx> Source for RecursivePathSource<'gctx> {
     }
 }
 
+/// [`PathBuf`] with extra metadata.
+#[derive(Clone, Debug)]
```

Comment:
> feat(source): wrap PathBuf with PathEntry

nit: as the lib is not intended as an API, this is more of a `refactor` than a `feat`

Reply:
Fixed in #14994